src/widgets/block.rs

```diff
@@ -176,13 +257,40 @@ impl<'a> Block<'a> {
     /// space is calculated based on the full width of the block, rather than the leftover width.
     ///
     /// You can provide various types as the title, including strings, string slices, borrowed
-    /// strings (`Cow<str>`), spans, or vectors of spans (`Vec<Span>`).
+    /// strings (`Cow<str>`), [spans](crate::text::Span), or vectors of [spans](crate::text::Span)
+    /// (`Vec<Span>`).
     ///
     /// By default, the titles will avoid being rendered in the corners of the block but will align
-    /// against the left or right edge of the block if there is no border on that edge.
+    /// against the left or right edge of the block if there is no border on that edge.  
+    /// The following demonstrates this behavior, notice the second title is one character off to
+    /// the left.
+    ///
+    /// ```plain
+    /// ┌With at least a left border───
+    ///
+    /// Without left border───
+    /// ```
     ///
     /// Note: If the block is too small and multiple titles overlap, the border might get cut off at
     /// a corner.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use ratatui::widgets::{Block, block::title::Title};
+    /// # use ratatui::layout::Alignment;
+    /// Block::default()
+    ///     .title("Title") // By default in the top left corner
+    ///     .title(Title::from("Right").alignment(Alignment::Right))
+    ///     .title(Title::from("Center").alignment(Alignment::Center));
+    /// ```
```

Comment:
It would be good to show a rendered version of this example in the doc. The note about how having multiple titles separated by a space would be more obvious if shown visually, (also showing that it's actually filled with a border character not a space)